### DISCUSSION TITLE
Payment App - Timeout after 5s, how to handle it?

### DISCUSSION THREAD

**Discussion by ChristopherDosin** [shopware/platform#3375] [App FAQ] [answered]
### What kind of question do you have about the app system?

Payment integration 

### Describe your question or feedback

Good Morning my friends,

It's about the implementation of a Payment handler for a Shopware App.
Within the Storefront the user can buy something and pay with the payment handler - It's a B2B payment App.

That being said, we're doing an API call to the payment handler and we're waiting for the response.
Since it's a B2B payment handler, there are quite some checks in the background from them against the customer who wants to pay with them.

Currently it takes around 3s.

However: Shopware set's a timeout to only 5s here ->
[NEXT-22775 - Add app timeout · shopware/shopware@1df7491 (github.com)](https://github.com/shopware/shopware/commit/1df749134cd4b0c87d54da87a27a458561226e18)

That being said:
If the API would take 6s for whatever reason, Shopware would throw a failure after 5s, the customer within the storefront would see a payment failure and Shopware mark the payment as failed.

But 1s later, for the payment handler it's authorized successfully, while the customer in the storefront thinks it's failed.

Can you point us here into the right direction, what's the reason for only 5s and how should we handle this?

**Comment by Christian-Rades** [comment]
Hey @ChristopherDosin We'll take care of this by increasing the timeout to 20 seconds.

**Reply by ChristopherDosin** [reply]
Thank you - But just still wondering:
It will never happen (hopefully :D), but what if it would take 21s? :)

The Payment handler would still be authorized, while the customer is getting a failed order within the checkout.

What I am missing here is the why it was set to 5s and how we should handle it, if it would happen for whatever reason.

Even though I understand of course that it should not happen to wait that long, speaking of the 5s - I am just wondering how to handle it if this case would happen.

**Comment by zfodor** [comment]
Hello @Christian-Rades. 
Do you have any updates on this?
I'm encountering the same issue, and I haven't noticed any changes related to the timeout in the latest versions either.

**Reply by FlorianKe** [reply, marked as answer]
Hey @ChristopherDosin

This problem has been fixed since 6.6.5.0 and 6.5.8.12. We have increased the value to 20 seconds.
